packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -138,6 +141,8 @@ const SUSPENSE_END_DATA = '/$';
 const SUSPENSE_PENDING_START_DATA = '$?';
 const SUSPENSE_FALLBACK_START_DATA = '$!';
 
+const INSERTION_MARKER = '%';
```

Comment:
It seems to me that all this insertion marker stuff is only needed because the React algorithm uses "insert before" instead of "insert after" - which is partially because that's what the only DOM APIs favored but also because we use linked lists that only go in one direction so we can't easily look backwards.

However, if it wasn't for that we'd have no need for these markers because we'd just insert after the last known node instead. It seems pretty inelegant to go through the hassle of mutating the DOM to insert visible placeholders when the algorithm should just be able to do that by itself.

Reply:
I updated the algo to not require any mutations. I think it is decent however lmk if the concerns of the speed of reading expando outweight the benefits of the new approach.

One thing I like about it is it also fixes portal insertion stability